### DISCUSSION TITLE
Renovate 25.x breaking changes with configuration

### DISCUSSION THREAD

**Discussion by hazendaz** [renovatebot/renovate#10227] [Request Help]
We use self hosted renovate.  Since 25.x, it doesn't seem to work the same with configuration as before.  It ignores our global renovate-config.js defaults and global ignores.  I've exhausted all attempts at making it work and rather shocked no one has reported the issue.  It doesn't seem to matter what 25.x.  If we drop back to 24.x everythign works.  We have ran this in full trace mode.  It shows all our configuration settings many times but ultimately decides to use other values.  Can someone clue us in here on what was so significant with 25.x that is causing such a difference in behaviour?

**Comment by rarkins** [comment]
If you describe in technical details about exactly how you configure Renovate then we can try to help.

**Reply by hazendaz** [reply]
The config:base doesn't work properly any longer with as prHourlyLimit2 and prConcurrentLimit20 seem to overtake prConcurrentLimit: 5 and prHourlyLimit: 0.  We were able to get around this by replacing config:base directly and removing those 2.  The logs show our settings throughout but finally it says it is calculating the values and it goes to pr hourly limit 2 and pr concurrent limit 20.  In earlier 25.8.x we last tried, this resulted in it creating up to 20 branches that were not expected and 2 pr as noted.  In the latest 25.33.x that we are now on, it sticks to just creating 2 branches at that 2 pr limit without us simply replacing config:base.  It doesn't make any sense as to why this is occurring but we were able to get around this one.

The bigger issue is that it is not adhering to package rules.  The one I'm directly trying to figure out is with commons-collections that exists in our repos incorrectly at specific version that was manually uploaded years ago.  We had it as regex in 24.x.  Upon 25.x, it immediately starts submitting all PRs for ignored packages.  It doesn't seem to matter if we use regex or just directly note the version.  Here is snippet of the setup.

packageRules: [
  {
      matchPackageNames: ['commons-collections:commons-collections'],
      allowedVersions: '!/4\.1$/'
  }

The logs again show this throughout as one to be excluded.  There are no warnings or errors.  Then it goes on to say 4.1 is valid update and submits the PR.

My next test which I haven't don't yet is to try setting that in renovate.json locally on the repo to confirm local ignore works.  However, we are trying to not allow local configuration at all to control updates in central way.  I can report back on that as soon as I have tested.  To this point though I have focused on the global config only and I'm at a total loss why it no longer works and struggling to understand what is so breaking in 25.x.  Also seems so many are asking for changes in config in 25.x so it seems others must be working.  We have been using renovate at this point since summer of 2020 and kept regularly up to date every few days.  The first real issue we have encountered where this seems to break is 25.x.  So we stayed on 24.x as long as we could but now that we don't see updates coming and 25.x is so far along, we feel its important we get here and stay up to date again but this is considerable issue.  To put this in perspective, we are self hostly upwards of 500 repos with the one global config.

**Comment by rarkins** [comment]
I think it's best if you try to narrow the problem down one at a time - Eg the packageRules one - and describe exactly how to reproduce it. Eg the minimal config.js and renovate.json which trigger the behaviour you're not expecting. The solution to the "narrow" problem may then turn out to be the root cause of all.

**Reply by hazendaz** [reply]
I was able to work around the issue.  It is an issue with how configuration is read since 25.x that differs from 24.x.  Basically 'config:base' cannot be directly used if there are overrides globally in the config.js (not the project level).  Everything that might conflict does.  So instead of using that, we simply defined what we needed that we were not overriding.  Further, we had to disable the apache commons-* preset as we could not properly override.  Only if we override on a downstream project does it work as expected with 'config:base' but that doesn't solve our need to do so globally without downstream configuration.  That tells me renovate is doing a read which changed between 24.x and 25.x and stops at first hit in configuration simply because loggign shows all the configuration and that is only thing that makes sense.  I suspect order changed on that configuratino read and that it probably never likely tried to merge all config together and best only allow the most accurate configuration item.  For example using the commons-* workarounds.  It would hit those fine (they are towards the end if used by 'config:base' in final configuration, but we have others that need excluded in commons-* so it was easy to test.  Anyway, would love to see this fixed but we are able to successfully work around it for time being simply by avoiding all presets at this point and just defining what we need.

**Reply by viceice** [reply]
It's because we deferred preset resolving and merging in bot config to support private presets.

That way package rules from preset override global settings.

So we generally don't recommend to use presets in bot config.
Instead we recommend a shared preset which is used in all repo configs.
